rpc/src/server/rpc_server.rs

```diff
@@ -0,0 +1,281 @@
+/// system with the server.
+pub struct RpcServer {
+    env: Arc<RpcServiceEnvironment>,
+    rpc_listen_address: SocketAddr,
+    log: Logger,
+    notify_rpc_server_tx: mpsc::SyncSender<NotifyRpcServerMsg>,
+    rpc_server_thread_handle: Option<RpcServerThreadHandle>,
+}
+
+impl RpcServer {
+    const NOTIFY_RPC_QUEUE_MAX_CAPACITY: usize = 50_000;
+
+    pub fn new(
+        sys: &riker::actors::ActorSystem,
+        log: Logger,
+        shell_channel: ShellChannelRef,
+        rpc_listen_address: SocketAddr,
+        tokio_executor: Handle,
+        persistent_storage: &PersistentStorage,
+        current_mempool_state_storage: CurrentMempoolStateStorageRef,
+        tezos_readonly_api: Arc<TezosApiConnectionPool>,
+        tezos_readonly_prevalidation_api: Arc<TezosApiConnectionPool>,
+        tezos_without_context_api: Arc<TezosApiConnectionPool>,
+        tezos_env: TezosEnvironmentConfiguration,
+        network_version: Arc<NetworkVersion>,
+        init_storage_data: &StorageInitInfo,
+        tezedge_is_enabled: bool,
+    ) -> Self {
+        let shared_state = Arc::new(RwLock::new(RpcCollectedState {
+            current_head: load_current_head(
+                persistent_storage,
+                &init_storage_data.chain_id,
+                &sys.log(),
+            ),
+        }));
+
+        let env = Arc::new(RpcServiceEnvironment::new(
+            sys.clone(),
+            Arc::new(tokio_executor),
+            shell_channel,
+            tezos_env,
+            network_version,
+            persistent_storage,
+            current_mempool_state_storage,
+            tezos_readonly_api,
+            tezos_readonly_prevalidation_api,
+            tezos_without_context_api,
+            init_storage_data.chain_id.clone(),
+            init_storage_data.genesis_block_header_hash.clone(),
+            shared_state,
+            init_storage_data.context_stats_db_path.clone(),
+            tezedge_is_enabled,
+            &sys.log(),
+        ));
+
+        let (notify_rpc_server_tx, notify_rpc_server_rx) =
+            mpsc::sync_channel(Self::NOTIFY_RPC_QUEUE_MAX_CAPACITY);
+
+        Self {
+            env,
+            rpc_listen_address,
+            log,
+            notify_rpc_server_tx,
+            rpc_server_thread_handle: Some(RpcServerThreadHandle::NotRunning(notify_rpc_server_rx)),
+        }
+    }
+
+    pub fn start(&mut self) -> Result<(), RpcServerSpawnError> {
+        if let Some(RpcServerThreadHandle::NotRunning(notify_rpc_server_rx)) =
+            self.rpc_server_thread_handle.take()
+        {
+            // 1. spawn mpsc queue receiver
+            // start to listen for incoming p2p connections and state machine processing
+            let notify_rpc_server_listener_thread = {
+                let env = self.env.clone();
+
+                std::thread::Builder::new()
+                    .name("rpc-notify-lstnr-thread".to_owned())
```

Comment:
Typo here, should be `"rpc-notify-listener-thread"`. Or is it for some reason to make name shorter?

Reply:
@binier @adonagy 
Ado, told me that, for some thread explorer, the linux uses just 15 char for thread name,
so I could rewrite it like `rpc-ntf-lstnr-thrd` :)

Reply:
Oh ok, makes sense. "thread" word is probably extra, so you could have just `rpc-notifier` or something like that.

Reply:
@binier 
right, so we have now:
`rpc-notifier`
`chainman-notifier`
and coming soon `ws-monitor-notifier`